substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -1879,3 +1872,9 @@ pub mod pallet {
 fn is_sorted_and_unique(list: &[u32]) -> bool {
 	list.windows(2).all(|w| w[0] < w[1])
 }
+
+/// Disabling limit calculated from the total number of validators in the active set. When reached
+/// no more validators will be disabled.
+pub fn disabling_limit(validators_len: usize) -> usize {
+	validators_len.saturating_sub(1) / 3
```

Comment:
I already explained why `OffendingValidatorsThreshold` is removed. If I understand correctly your concern here is why the disabling limit is hardcoded instead of parametrised.
At this point I think the disabling limit should be unified. `OffendingValidatorsThreshold` was set to different values in the mock and prod runtimes. I think this is confusing. Disabling limit is a hard limit. If someone wants it changed , he/she can change this function with a PR and a good motivation.
Do you agree?

Reply:
Adding reasoning for this limit in the docs would be good.

Our byzantine assumptions require, that no more than a third of the nodes are faulty. Hence if we ever ended up disabling more nodes than that (which implies that we consider them faulty), we know that our assumptions are off. We have to assume there is some bug present and should stop disabling more nodes.

Reply:
Those are good points, thanks! As a general rule, we try to avoid hard coded magic numbers in the code base. There are a couple of reasons for this:

1. It's hard to find the numbers and what they mean;
2. It's not very developer friendly for other projects that use this pallet. This pallet is used by other projects which may have different underlying security assumption, consensus etc. In this case, they would have to fork the pallet just to change this value and carry the fork along as we update the pallet.

I understand the concern with having different values in the mock etc. But we can have a [`Hooks::integrity_test()`](https://paritytech.github.io/polkadot-sdk/master/frame_support/traits/trait.Hooks.html#method.integrity_test) check that ensures the value is 1/3 at each construct runtime (tests with externalities included).

This said, I'm still inclined to have the `1/3` as a storage item, construct runtime const or as a [storage parameter type](https://paritytech.github.io/polkadot-sdk/master/frame_support/macro.parameter_types.html).

Reply:
Agreed, that makes a lot of sense!

Reply:
Extracting the disabling strategy behind a trait doesn't require a storage item for the disabling limit in my opinion. It can be a constant in the trait or even a method doing something more complicated if necessary,